src/main/kotlin/lotto/domain/PurchasedLottos.kt

```diff
@@ -0,0 +1,26 @@
+import lotto.domain.Lotto
+import lotto.domain.LottoNumber
+import lotto.domain.Rank
+
+class PurchasedLottos(private val tickets: List<Lotto>) {
+    // Lottos 라는 객체는 구입한 로또들의 집합.
+    init {
+        require(tickets.isNotEmpty()) { "로또는 최소 1개 이상 구매해야 합니다." }
+    }
+
+    fun size(): Int = tickets.size
+
+    val lottos: List<Lotto>
+        get() = tickets.toList()
+
+    fun matchNumber(
+        winningNumber: Lotto,
+        bonusBall: LottoNumber,
+    ): Map<Rank, Int> {
+        return tickets
+            .map { it.match(winningNumber, bonusBall) }
+            .map { (matchCount, matchBonus) -> Rank.from(matchCount, matchBonus) }
```

Comment:
Lotto.match의 결과로 `Rank`를 반환하면 더 깔끔할 것 같은데, 어떻게 생각하시나요? 😃

```suggestion
            .map { it.match(winningNumber, bonusBall) }
```

Reply:
여기서 좀 더 세분화해서 결과에 대한 data클래스를 나누려고 MatchResult 객체를 빼서 뒀는데, 말씀하신 내용 생각해보니 
해당 부분에 대한 책임을 rank로 대신할 수 있을것 같다는 생각이 드네요!